watertap/tools/oli_api/flash.py

```diff
@@ -0,0 +1,494 @@
+            )
+
+            self.water_analysis_input_list.append(
+                {
+                    "group": get_charge_group(charge),
+                    "name": name,
+                    "unit": conc[1],
+                    "value": conc[0],
+                    "charge": charge,
+                }
+            )
+
+        for k, v in self.water_analysis_properties.items():
+            if v["value"] is not None:
+                if isinstance(v["value"], list):
+                    self.water_analysis_properties[k]["value"] = v["value"][0]
+                if all(k != i["name"] for i in self.water_analysis_input_list):
+                    self.water_analysis_input_list.append(v)
+
+        input_list = deepcopy(self.water_analysis_input_list)
+        return input_list
+
+    def _oli_units(self, conc, unit):
+        """
+        Converts concentrations between specified units.
+
+        :param conc: concentration value for a solute
+        :param unit: concentration unit for a solute
+
+        :return converted_value: tuple with converted concentration and OLI unit string
+        """
+        to_units = input_unit_set["molecularConcentration"]
+        converted_value = (
+            pyunits.convert_value(conc, unit, to_units["pyomo_unit"]),
+            to_units["oli_unit"],
+        )
+        return converted_value
+
+    def _set_prescaling_calculation_mode(self, use_scaling_rigorous):
+        """
+        Sets prescaling computation method based on argument.
+
+        :param use_scaling_rigorous: boolean indicating desired state of 'rigorous' and 'estimated' optional properties
+        """
+
+        if bool(use_scaling_rigorous) == bool(
+            self.optional_properties["prescalingTendenciesRigorous"]
+        ):
+            return
+        new_values = {
+            k: not v for k, v in self.optional_properties.items() if "prescaling" in k
+        }
+        self.optional_properties.update(new_values)
+
+    def build_flash_calculation_input(
+        self,
+        flash_method,
+        state_vars,
+        water_analysis_output=None,
+        inflows_phase="total",
+        use_scaling_rigorous=True,
+    ):
+        """
+        Builds a base dictionary required for OLI flash analysis.
+
+        :param flash_method: string name of OLI flash flash_method to use
+        :param state_vars: dictionary containing solutes, temperatures, pressure, and units
+        :param water_analysis_output: dictionary to extract inflows from (required if not wateranalysis flash)
+        :param inflows_phase: string indicating desired phase of inflows
+        :param use_scaling_rigorous: boolean switch to use estimated or rigorous solving for prescaling metrics
+
+        :return inputs: dictionary containing inputs for specified OLI flash analysis
+        """
+
+        inputs = {"params": {}}
+
+        if flash_method == "wateranalysis":
```

Comment:
OLI presents "wateranalysis" as a separate function from "isothermal flash", but it seems that you're combining the two into a single "Flash" class. Am I interpreting that correctly? Are there reasons for doing it that way rather than maintaining two separate classes to emulate how OLI presents these capabilities in their API documentation?